### PR TITLE
Management Center Support [API-501, API-502]

### PR DESCRIPTION
**Pull request by yuce**
- Adds Management Center 4.2 and 4.0.3 support
- Adds binary and text statistics.
- Cluster connected events carry the initial connection address.

### CODE REVIEW

**Review by yemreinci — COMMENTED** [review]
I haven't look all the code yet but have you tested with MC 4.0.3? MC 4.0.3 doesn't use the new binary blob, it depends on the `clientAttributes` parameter. Java and Node.js clients currently send metrics in both parameters for backwards compatibility.

**Comment by yuce** [comment]
> 
> 
> I haven't looked at all the code yet but have you tested with MC 4.0.3? MC 4.0.3 doesn't use the new binary blob, it depends on the `clientAttributes` parameter. Java and Node.js clients currently send metrics in both parameters for backwards compatibility.

Thanks, good point. I've added support for text metrics and tested against Hz/MC 4.0.3

**Review by yemreinci — APPROVED** [review]
Approving with 2 minor comments.

File: internal/common.go [hunk 1]
```diff
@@ -14,6 +14,13 @@
  * limitations under the License.
  */
 
-package proto
+package internal
 
-const ClientType = "GO"
+const (
+	// ClientType is used in the Management
+	ClientType = "GOO"
```
[the remark this answers is not in the document]
Comment (yuce) [reply internal/common.go:21]: That ID is used by the MC do determine the client type. I've changed it to `GO` before, which was wrong. I guess we're stuck with `GOO` :)

**Pull request merged**